src/dashboard/apps/consent/helpers.py

```diff
@@ -76,3 +83,77 @@ def send_mail(recipients: list, template_id: int, email_data: dict) -> int | Non
         # fail silently and send a sentry log
         sentry_sdk.capture_exception(e)
         return 0
+
+
+def duplicate_expiring_consents():
+    """Duplicate expiring consents.
+
+    date de debut du nouveau consentement = date de fin du consentement précédent
+    date de fin du nouveau consentement =
+        si CONSENT_UPCOMING_DAYS_LIMIT => consent_end_date() + 1 year
+        sinon => consent_end_date()
+    """
+    """Duplicate expiring consents to ensure continuity.
+
+    Duplicate expiring consents to ensure continuity for consents that are nearing
+    their expiration date and have not been duplicated yet.
+    The function identifies consents that are set to expire shortly and generates
+    new consents starting immediately after the expiry date of the current consent,
+    with an extended validity period.
+
+    - consents must be duplicated if end date is <= now + CONSENT_NUMBER_DAYS_END_DATE
+    - duplicated consent date:
+        - start date of new consent is end date of current consent
+        - end date of new consent is calculate from CONSENT_NUMBER_DAYS_END_DATE
+
+    Returns:
+        List[Consent]: A list of newly created Consent objects that have been duplicated
+         from the expiring consents.
+    """
+    # calculate the date from which to check
+    now = timezone.now()
+    end_of_today = now.replace(hour=23, minute=59, second=59, microsecond=999999)
+    checking_date = end_of_today + timedelta(days=settings.CONSENT_UPCOMING_DAYS_LIMIT)
```

Comment:
```suggestion
    # calculate the date from which to check
    now = timezone.now()
    end_of_today = now.replace(hour=23, minute=59, second=59, microsecond=999999)
    checking_date = end_of_today + timedelta(days=settings.CONSENT_UPCOMING_DAYS_LIMIT)
```
It would be nice to define an helper in `apps.consent.utils` for this